contribs/gnofaucet/gh.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/v64/github"
+)
+
+// getGithubMiddleware sets up authentication middleware for GitHub OAuth.
+// If clientID and secret are empty, the middleware does nothing.
+//
+// Parameters:
+// - clientID: The OAuth client ID issued by GitHub when registering the application.
+// - secret: The OAuth client secret used to securely authenticate API requests.
+// - cooldown: A cooldown duration to prevent several claims from the same user.
+//
+// GitHub OAuth applications require a client ID and secret to authenticate users securely.
+// These credentials are obtained when registering an application on GitHub at:
+// https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authenticating-to-the-rest-api-with-an-oauth-app#registering-your-app
+func getGithubMiddleware(clientID, secret string, coolDownLimiter *CooldownLimiter) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				// github Oauth flow is enabled
+				if secret == "" || clientID == "" {
```

Comment:
If this middleware is spawned, there is no question if these are set or not

Reply:
Now as we have separate commands the middleware is only spawned if github command is used f984a7b50f01ad9950a0411b1762d211d1679e1f